radixsort.c

```diff
@@ -0,0 +1,167 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+
+#include "radixsort.h"
+#include "ordenacaomacros.h"
+
+void radixsort(Item *v, int l, int n){
+  // v = vetor de números
+  // n = quantidade de números no vetor
+  // ex.: [0, 1, 2, 3] -> n=4
+
+  int aux, aux1, maxdigitos = 1;
+  n++;
+
+  // Calcula a maior quantidade de dígitos de um
+  // único número no vetor
+  for (int i = l; i < n; i++){
+    int div = 10;
+    int maior = 1;
+    while (v[i] / div != 0){
+      div *= 10;
+      maior++;
+    }
+    if (maior > maxdigitos){
+      maxdigitos = maior;
+    }
+  }
+
+  for (int j = 0; j < maxdigitos; j++){
+    int z = 0, u = 0, d = 0, t = 0, q = 0, c = 0, s = 0, st = 0, o = 0, nv = 0;
+
+    int *zero = malloc(n * sizeof(int));
```

Comment:
nesta implementação o `n` equivale ao valor de `r`  e, portanto, não representa o tamanho do vetor sendo ordenado (pode ser super-estimado), há dois caminhos para resolver isso:
1. Adaptar neste ponto o tamanho de cada vetor para o cálculo correto `(n-l+1)`, ou;
2. trocar o parâmetro do vetor de `n` para `r` e declarar uma variável `n` com o tamanho do vetor logo no início da função, e atualizar todos os usos de `n` na função